apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs

```diff
@@ -36,6 +36,44 @@ namespace Google.Cloud.Spanner.V1
     /// </summary>
     public sealed partial class SessionPool
     {
+        /// <summary>
+        /// This struct contains information required to identify the database sessions.
+        /// It will be used as key in <see cref="_targetedPools"/>.
+        /// </summary>
+        internal readonly struct SessionPoolInfo
+        {
+            /// <summary>
+            /// Database name of the associated <see cref="TargetedSessionPool"/>.
+            /// </summary>
+            internal DatabaseName DatabaseName { get; }
+
+            /// <summary>
+            /// Database role of the associated <see cref="TargetedSessionPool"/>.
+            /// </summary>
+            internal string DatabaseRole { get; }
+
+            internal SessionPoolInfo(DatabaseName databaseName, string dataabaseRole)
+            {
+                DatabaseName = databaseName;
+                DatabaseRole = dataabaseRole;
+            }
+
+            /// <inheritdoc/>
+            public override string ToString() => $"Database Name = {DatabaseName};Database role = {DatabaseRole}";
+
+            /// <inheritdoc/>
+            public override int GetHashCode() => ToString().GetHashCode();
+
+            /// <inheritdoc/>
+            public override bool Equals(object obj)
+            {
+                if (obj is not SessionPoolInfo)
+                {
+                    return false;
+                }
+                return obj.ToString() == ToString();
```

Comment:
We shouldn't perform string formatting every time. See https://github.com/googleapis/gax-dotnet/blob/main/Google.Api.Gax/GaxEqualityHelpers.cs for helpers for this.